Error installing rPackages.rhdf5

## Issue description
Installation of `rPackages.rhdf5` fails using both my default channel and current `master` branch of `NixOS/nixpkgs` (18.03-beta-14018-g8cf4a4725c1).


### Steps to reproduce
```
nix-build -E 'with import <nixpkgs> {}; rWrapper.override { packages = [rPackages.rhdf5]; }';
```

<details><summary>Full output</summary><p>

```
these derivations will be built:
  /nix/store/99z8035b3jdj4aikikdc4l5srg0xc16q-r-rhdf5-2.22.0.drv
  /nix/store/zzax7a0vskgzqg221bc8pzx6b114w4gi-R-3.5.1-wrapper.drv
building '/nix/store/99z8035b3jdj4aikikdc4l5srg0xc16q-r-rhdf5-2.22.0.drv'...
unpacking sources
unpacking source archive /nix/store/wksdxj17l3qdfy5aa7vilaj73ab925dd-rhdf5_2.22.0.tar.gz
source root is rhdf5
setting SOURCE_DATE_EPOCH to timestamp 1509405845 of file rhdf5/src/wrap.c
patching sources
configuring
building
running tests
installing
* installing *source* package 'rhdf5' ...
checking for gcc... /nix/store/yz6kinf4ia19r1c14yirl6x4ciwgzk67-gcc-wrapper-7.3.0/bin/cc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether /nix/store/yz6kinf4ia19r1c14yirl6x4ciwgzk67-gcc-wrapper-7.3.0/bin/cc accepts -g... yes
checking for /nix/store/yz6kinf4ia19r1c14yirl6x4ciwgzk67-gcc-wrapper-7.3.0/bin/cc option to accept ISO C89... none needed
checking whether we are using the GNU C++ compiler... yes
checking whether g++ accepts -g... yes
untarring hdf5small.tgz ...
building the bundled hdf5 library...
checking for a BSD-compatible install... /nix/store/n7qp8pffvcb5ff52l2nrc3g2wvxfrk75-coreutils-8.29/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /nix/store/n7qp8pffvcb5ff52l2nrc3g2wvxfrk75-coreutils-8.29/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether to enable maintainer-specific portions of Makefiles... no
checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
checking shell variables initial values... done
checking if basename works... yes
checking if xargs works... yes
checking for cached host... none
checking for config x86_64-unknown-linux-gnu... no
checking for config x86_64-unknown-linux-gnu... no
checking for config unknown-linux-gnu... no
checking for config unknown-linux-gnu... no
checking for config x86_64-linux-gnu... no
checking for config x86_64-linux-gnu... no
checking for config x86_64-unknown... no
checking for config linux-gnu... found
compiler '/nix/store/yz6kinf4ia19r1c14yirl6x4ciwgzk67-gcc-wrapper-7.3.0/bin/cc' is GNU gcc-7.3.0
./configure: ./config/gnu-flags: line 175: hostname: not found
CC_BASENAME is cc.
No match to get fc_version_info for
./configure: ./configure.lineno: line 3956: hostname: not found
checking for gcc... /nix/store/yz6kinf4ia19r1c14yirl6x4ciwgzk67-gcc-wrapper-7.3.0/bin/cc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether /nix/store/yz6kinf4ia19r1c14yirl6x4ciwgzk67-gcc-wrapper-7.3.0/bin/cc accepts -g... yes
checking for /nix/store/yz6kinf4ia19r1c14yirl6x4ciwgzk67-gcc-wrapper-7.3.0/bin/cc option to accept ISO C89... none needed
checking for style of include used by make... GNU
checking dependency style of /nix/store/yz6kinf4ia19r1c14yirl6x4ciwgzk67-gcc-wrapper-7.3.0/bin/cc... gcc3
checking if unsupported combinations of configure options are allowed... no
checking if Fortran interface enabled... no
checking whether we are using the GNU C++ compiler... yes
checking whether g++ accepts -g... yes
checking dependency style of g++... gcc3
checking how to run the C++ preprocessor... g++ -E
checking if c++ interface enabled... no
checking for perl... no
checking whether make sets $(MAKE)... (cached) yes
checking for tr... /nix/store/n7qp8pffvcb5ff52l2nrc3g2wvxfrk75-coreutils-8.29/bin/tr
checking if srcdir= and time commands work together... no
checking how to print strings... printf
checking for a sed that does not truncate output... /nix/store/4qq502jmmwqrvr9y42m8kax686ppm6mh-gnused-4.5/bin/sed
checking for grep that handles long lines and -e... /nix/store/45ilvajpznqfakvj2ab5f4wcjcv1f3g7-gnugrep-3.1/bin/grep
checking for egrep... /nix/store/45ilvajpznqfakvj2ab5f4wcjcv1f3g7-gnugrep-3.1/bin/grep -E
checking for fgrep... /nix/store/45ilvajpznqfakvj2ab5f4wcjcv1f3g7-gnugrep-3.1/bin/grep -F
checking for ld used by /nix/store/yz6kinf4ia19r1c14yirl6x4ciwgzk67-gcc-wrapper-7.3.0/bin/cc... ld
checking if the linker (ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... nm
checking the name lister (nm) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 1572864
checking whether the shell understands some XSI constructs... yes
checking whether the shell understands "+="... no
checking how to convert x86_64-unknown-linux-gnu file names to x86_64-unknown-linux-gnu format... func_convert_file_noop
checking how to convert x86_64-unknown-linux-gnu file names to toolchain format... func_convert_file_noop
checking for ld option to reload object files... -r
checking for objdump... objdump
checking how to recognize dependent libraries... pass_all
checking for dlltool... dlltool
checking how to associate runtime and link libraries... printf %s\n
checking for archiver @FILE support... @
checking for strip... strip
checking for ranlib... ranlib
checking command to parse nm output from /nix/store/yz6kinf4ia19r1c14yirl6x4ciwgzk67-gcc-wrapper-7.3.0/bin/cc object... ok
checking for sysroot... no
./configure: ./configure.lineno: line 1: /usr/bin/file: not found
checking for mt... no
checking if : is a manifest tool... no
checking how to run the C preprocessor... /nix/store/yz6kinf4ia19r1c14yirl6x4ciwgzk67-gcc-wrapper-7.3.0/bin/cc -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if /nix/store/yz6kinf4ia19r1c14yirl6x4ciwgzk67-gcc-wrapper-7.3.0/bin/cc supports -fno-rtti -fno-exceptions... no
checking for /nix/store/yz6kinf4ia19r1c14yirl6x4ciwgzk67-gcc-wrapper-7.3.0/bin/cc option to produce PIC... -fPIC -DPIC
checking if /nix/store/yz6kinf4ia19r1c14yirl6x4ciwgzk67-gcc-wrapper-7.3.0/bin/cc PIC flag -fPIC -DPIC works... yes
checking if /nix/store/yz6kinf4ia19r1c14yirl6x4ciwgzk67-gcc-wrapper-7.3.0/bin/cc static flag -static works... no
checking if /nix/store/yz6kinf4ia19r1c14yirl6x4ciwgzk67-gcc-wrapper-7.3.0/bin/cc supports -c -o file.o... yes
checking if /nix/store/yz6kinf4ia19r1c14yirl6x4ciwgzk67-gcc-wrapper-7.3.0/bin/cc supports -c -o file.o... (cached) yes
checking whether the /nix/store/yz6kinf4ia19r1c14yirl6x4ciwgzk67-gcc-wrapper-7.3.0/bin/cc linker (ld) supports shared libraries... yes
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking for shl_load... no
checking for shl_load in -ldld... no
checking for dlopen... no
checking for dlopen in -ldl... yes
checking whether a program can dlopen itself... yes
checking whether a statically linked program can dlopen itself... yes
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... no
checking whether to build static libraries... yes
checking if we should install only statically linked executables... no
checking if -Wl,-rpath should be used to link shared libs in nondefault directories... yes
checking make... checking how make searches directories... VPATH=DIR1 DIR2 ...
checking whether make will build with undefined variables... yes
checking for production mode... production
checking for ceil in -lm... yes
checking for ANSI C header files... (cached) yes
checking whether time.h and sys/time.h may both be included... yes
checking for difftime... yes
checking for gettimeofday... yes
checking sys/resource.h usability... yes
checking sys/resource.h presence... yes
checking for sys/resource.h... yes
checking sys/time.h usability... yes
checking sys/time.h presence... yes
checking for sys/time.h... yes
checking for unistd.h... (cached) yes
checking sys/ioctl.h usability... yes
checking sys/ioctl.h presence... yes
checking for sys/ioctl.h... yes
checking for sys/stat.h... (cached) yes
checking sys/socket.h usability... yes
checking sys/socket.h presence... yes
checking for sys/socket.h... yes
checking for sys/types.h... (cached) yes
checking stddef.h usability... yes
checking stddef.h presence... yes
checking for stddef.h... yes
checking setjmp.h usability... yes
checking setjmp.h presence... yes
checking for setjmp.h... yes
checking features.h usability... yes
checking features.h presence... yes
checking for features.h... yes
checking for stdint.h... (cached) yes
checking io.h usability... no
checking io.h presence... no
checking for io.h... no
checking winsock.h usability... no
checking winsock.h presence... no
checking for winsock.h... no
checking sys/timeb.h usability... yes
checking sys/timeb.h presence... yes
checking for sys/timeb.h... yes
checking if configure should try to set up large file support... yes
checking for special C compiler options needed for large files... no
checking for _FILE_OFFSET_BITS value needed for large files... no
checking if large (64-bit) files are supported on this system.... yes
checking for lseek64... yes
checking for fseeko64... yes
checking for ftello64... yes
checking for ftruncate64... yes
checking for fseeko... yes
checking for ftello... yes
checking for stat64... yes
checking for fstat64... yes
checking for off_t... yes
checking for size_t... yes
checking for ssize_t... yes
checking for ptrdiff_t... yes
checking whether byte ordering is bigendian... no
checking size of char... 1
checking size of short... 2
checking size of int... 4
checking size of unsigned... 4
checking size of long... 8
checking size of long long... 8
checking size of __int64... 0
checking size of float... 4
checking size of double... 8
checking size of long double... 16
checking size of int8_t... 1
checking size of uint8_t... 1
checking size of int_least8_t... 1
checking size of uint_least8_t... 1
checking size of int_fast8_t... 1
checking size of uint_fast8_t... 1
checking size of int16_t... 2
checking size of uint16_t... 2
checking size of int_least16_t... 2
checking size of uint_least16_t... 2
checking size of int_fast16_t... 8
checking size of uint_fast16_t... 8
checking size of int32_t... 4
checking size of uint32_t... 4
checking size of int_least32_t... 4
checking size of uint_least32_t... 4
checking size of int_fast32_t... 8
checking size of uint_fast32_t... 8
checking size of int64_t... 8
checking size of uint64_t... 8
checking size of int_least64_t... 8
checking size of uint_least64_t... 8
checking size of int_fast64_t... 8
checking size of uint_fast64_t... 8
checking size of size_t... 8
checking size of ssize_t... 8
checking size of ptrdiff_t... 8
checking size of off_t... 8
checking size of off64_t... 8
checking if dev_t is scalar... yes
checking for dmalloc library... suppressed
checking zlib.h usability... no
checking zlib.h presence... no
checking for zlib.h... no
checking for compress2 in -lz... yes
checking for compress2... yes
checking for szlib... suppressed
checking for pthread... suppressed
checking for thread safe support... no
checking for tm_gmtoff in struct tm... yes
checking for __tm_gmtoff in struct tm... no
checking for global timezone variable... yes
checking whether struct tm is in sys/time.h or time.h... time.h
checking for struct tm.tm_zone... yes
checking for struct timezone... yes
checking whether gettimeofday() gives timezone... yes
checking for st_blocks in struct stat... yes
checking for _getvideoconfig... no
checking for gettextinfo... no
checking for GetConsoleScreenBufferInfo... no
checking for _scrsize... no
checking for ioctl... yes
checking for struct videoconfig... no
checking for struct text_info... no
checking for TIOCGWINSZ... yes
checking for TIOCGETD... yes
checking for alarm... yes
checking for BSDgettimeofday... no
checking for fork... yes
checking for frexpf... yes
checking for frexpl... yes
checking for gethostname... yes
checking for getpwuid... yes
checking for getrusage... yes
checking for lstat... yes
checking for rand_r... yes
checking for random... yes
checking for setsysinfo... no
checking for signal... yes
checking for longjmp... yes
checking for setjmp... yes
checking for siglongjmp... yes
checking for sigsetjmp... no
checking for sigprocmask... yes
checking for snprintf... yes
checking for srandom... yes
checking for strdup... yes
checking for symlink... yes
checking for system... yes
checking for tmpfile... yes
checking for vasprintf... yes
checking for waitpid... yes
checking for vsnprintf... yes
checking if vsnprintf returns correct value... yes
checking if lone colon can be used as an argument... yes
checking for an ANSI C-conforming const... yes
checking for inline... inline
checking for __attribute__ extension... yes
checking for __func__ extension... yes
checking for __FUNCTION__ extension... yes
checking for C99 designated initialization support... yes
checking how to print long long... %ld and %lu
checking Threads support system scope... yes
checking for gpfs... suppressed
checking for debug flags... none
checking whether function stack tracking is enabled... no
checking whether metadata trace file code is enabled... no
checking for API tracing... no
checking for instrumented library... no
checking whether to clear file buffers... yes
checking whether a memory checking tool will be used... no
checking for parallel support files... skipped
checking for I/O filters... all (shuffle,fletcher32,nbit,scaleoffset)
checking for Default Virtual File Driver definition... yes
checking for Direct Virtual File Driver support... no
checking whether exception handling functions is checked during data conversions... yes
checking whether data accuracy is guaranteed during data conversions... yes
checking if converting denormalized floating-point values is possible... yes
checking if converting floating-point values to long long is not working... false
checking if the machine has window style path name... no
checking if converting from long double to integers is accurate... yes
checking if converting from long double to integers works... yes
checking if accurately converting from integers to long double... yes
checking if accurately converting unsigned long to float values... yes
checking if accurately converting unsigned long long to floating-point values... yes
checking if accurately roundup converting floating-point to unsigned long long values... yes
checking if right maximum converting floating-point to unsigned long long values... yes
checking if correctly converting long double to unsigned int values... yes
checking if compiling unsigned long long to floating-point typecasts work... yes
checking if compiling long long to floating-point typecasts work... yes
checking if converting unsigned long long to long double with precision... yes
checking if overflows normally converting floating-point to integer values... yes
checking if using special algorithm to convert long double to (unsigned) long values... no
checking if using special algorithm to convert (unsigned) long to long double values... no
checking if correctly converting long double to (unsigned) long long values... yes
checking if correctly converting (unsigned) long long to long double values... yes
checking if bad code for log2 routine is generated... no
./configure: ./configure.lineno: line 29478: hostname: not found
checking if high level library is enabled... yes
checking additional programs should be built... no
checking if deprecated public symbols are available... yes
checking which version of public symbols to use by default... v18
checking Whether to perform strict file format checks... no
checking Whether to have library information embedded in the executables... yes
checking Whether to enable the h5dump packed bits feature... yes
checking if alignment restrictions are strictly enforced... no
configure: creating ./config.status
configure: creating ./config.lt
config.lt: creating libtool
config.status: creating src/libhdf5.settings
config.status: creating Makefile
config.status: creating src/Makefile
config.status: creating src/H5config.h
config.status: executing default-1 commands
creating src/H5pubconf.h
./config.status: line 1769: /bin/mv: No such file or directory
Post process src/libhdf5.settings
config.status: executing depfiles commands
config.status: executing libtool commands
chmod: cannot access 'tools/misc/h5cc': No such file or directory
            SUMMARY OF THE HDF5 CONFIGURATION
            =================================

General Information:
-------------------
                   HDF5 Version: 1.8.7
                  Configured on: Mon Jul  2 15:00:46 UTC 2018
                  Configured by: nixbld@
                 Configure mode: production
                    Host system: x86_64-unknown-linux-gnu
              Uname information: Linux localhost 4.14.46 #1-NixOS SMP Wed May 30 10:19:59 UTC 2018 x86_64 GNU/Linux
                       Byte sex: little-endian
                      Libraries:
             Installation point: /build/rhdf5/src/hdf5/hdf5

Compiling Options:
------------------
               Compilation Mode: production
                     C Compiler: /nix/store/yz6kinf4ia19r1c14yirl6x4ciwgzk67-gcc-wrapper-7.3.0/bin/cc ( gcc (GCC) 7.3.0)
                         CFLAGS: -g -O2
                      H5_CFLAGS:   -ansi -pedantic -Wall -W -Wundef -Wshadow -Wpointer-arith -Wbad-function-cast -Wcast-qual -Wcast-align -Wwrite-strings -Wconversion -Waggregate-return -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations -Wredundant-decls -Wnested-externs -Winline -w -O -fomit-frame-pointer -finline-functions
                      AM_CFLAGS:
                       CPPFLAGS:
                    H5_CPPFLAGS: -D_POSIX_SOURCE   -DNDEBUG -UH5_DEBUG_API
                    AM_CPPFLAGS: -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_BSD_SOURCE
               Shared C Library: no
               Static C Library: yes
  Statically Linked Executables: no
                        LDFLAGS:
                     H5_LDFLAGS:
                     AM_LDFLAGS:
                Extra libraries:  -lz -lm
                       Archiver: ar
                         Ranlib: ranlib
              Debugged Packages:
                    API Tracing: no

Languages:
----------
                        Fortran: no

                            C++: no

Features:
---------
                  Parallel HDF5: no
             High Level library: yes
                   Threadsafety: no
            Default API Mapping: v18
 With Deprecated Public Symbols: yes
         I/O filters (external):
         I/O filters (internal): shuffle,fletcher32,nbit,scaleoffset
                            MPE: no
                     Direct VFD: no
                        dmalloc: no
Clear file buffers before write: yes
           Using memory checker: no
         Function Stack Tracing: no
                           GPFS: no
      Strict File Format Checks: no
   Optimization Instrumentation: no
       Large File Support (LFS): yes
             H5dump Packed Bits: yes
+ for d in src
+ test src '!=' .
+ cd src
+ make lib
make[1]: Entering directory '/build/rhdf5/src/hdf5/src'
make[2]: Entering directory '/build/rhdf5/src/hdf5/src'
  CC     H5.lo
In file included from H5private.h:29:0,
                 from H5.c:24:
H5public.h:31:10: fatal error: H5pubconf.h: No such file or directory
 #include "H5pubconf.h"  /*from configure                             */
          ^~~~~~~~~~~~~
compilation terminated.
make[2]: *** [Makefile:955: H5.lo] Error 1
make[2]: Leaving directory '/build/rhdf5/src/hdf5/src'
make[1]: *** [Makefile:1326: lib] Error 1
make[1]: Leaving directory '/build/rhdf5/src/hdf5/src'
+ exit 1
make: *** [Makefile:940: lib] Error 1
configure:    HDF5_INCLUDE=hdf5/src
configure:    HDF5_LIB=hdf5/src/.libs/libhdf5ForBioC.a
configure: creating ./config.status
config.status: creating src/Makevars
** libs
/nix/store/yz6kinf4ia19r1c14yirl6x4ciwgzk67-gcc-wrapper-7.3.0/bin/cc -I"/nix/store/nr5sj33z4bqs74yxwam8yb81gbfhcdw5-R-3.5.1/lib/R/include" -DNDEBUG     -Ihdf5/src -fpic  -g -O2  -c H5.c -o H5.o
In file included from hdf5/src/hdf5.h:24:0,
                 from myhdf5.h:7,
                 from H5.h:8,
                 from H5.c:1:
hdf5/src/H5public.h:31:10: fatal error: H5pubconf.h: No such file or directory
 #include "H5pubconf.h"  /*from configure                             */
          ^~~~~~~~~~~~~
compilation terminated.
make: *** [/nix/store/nr5sj33z4bqs74yxwam8yb81gbfhcdw5-R-3.5.1/lib/R/etc/Makeconf:159: H5.o] Error 1
ERROR: compilation failed for package 'rhdf5'
* removing '/nix/store/7340sksp7jalvgw6k981fjy33alsc5k4-r-rhdf5-2.22.0/library/rhdf5'
builder for '/nix/store/99z8035b3jdj4aikikdc4l5srg0xc16q-r-rhdf5-2.22.0.drv' failed with exit code 1
cannot build derivation '/nix/store/zzax7a0vskgzqg221bc8pzx6b114w4gi-R-3.5.1-wrapper.drv': 1 dependencies couldn't be built
error: build of '/nix/store/zzax7a0vskgzqg221bc8pzx6b114w4gi-R-3.5.1-wrapper.drv' failed
```

</p></details>

I was able to track the main cause of the problem down to this line:
```
./config.status: line 1769: /bin/mv: No such file or directory
```

A file that is dynamically created during installation contains a hard coded `/bin/mv` instead of plain `mv`. This causes the missing headers and subsequently the failing compilation.
I created a patch for this and will submit a pull request shortly.

## Technical details
 - system: `"x86_64-linux"`
 - host os: `Linux 4.14.46, NixOS, 18.03.132545.c40fdcd26c2 (Impala)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.0.2`
 - channels(root): `"nixos-18.03.132545.c40fdcd26c2"`
 - channels(user): `"nixpkgs-18.09pre140731.c29d2fde74d"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs`

--- a/pkgs/development/r-modules/default.nix
+++ b/pkgs/development/r-modules/default.nix
@@ -749,6 +749,10 @@ let
       patches = [ ./patches/BayesXsrc.patch ];
     });
 
+    rhdf5 = old.rhdf5.overrideDerivation (attrs: {
+      patches = [ ./patches/rhdf5.patch ];
+    });
+
     rJava = old.rJava.overrideDerivation (attrs: {
       preConfigure = ''
         export JAVA_CPPFLAGS=-I${pkgs.jdk}/include/

--- a/pkgs/development/r-modules/patches/rhdf5.patch
+++ b/pkgs/development/r-modules/patches/rhdf5.patch
@@ -1,0 +1,12 @@
+diff --git a/configure b/configure
+index e3e21e8..3d947b6 100755
+--- a/configure
++++ b/configure
+@@ -2859,6 +2859,7 @@ fi;
+ 
+ echo "building the bundled hdf5 library...";
+ cd ${BASEPBNAME};
++sed -i 's#/bin/mv#mv#' configure
+ ./configure  --with-pic --enable-shared=no CXX="${CXX}" CXXFLAGS="${CXXFLAGS}" CC="${CC}" CFLAGS="${CFLAGS}" F77="${F77}"
+ $MAKE lib
+ cd ../../

